src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,30 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.RoutingEngine;
+import java.util.regex.*;
+
+
+public class parsingEngine {
+ 	public void ParseCommands(String message){
+ 		BotRequest request;
+
+ 		// ^ to make sure we always start at the beginning of a string without whitespace.
+ 		// / to detect our command syntax.
+ 		// (?P<command_name>\w+) as a required parameter capture group that is made up of at least 1 alphanumeric character
+ 		// \s as a white space chracter. : space, tab, newline, carriage return, vertical tab. 
```

Comment:
Is this describing any particular part of the regex?

There are no comments about the variable number of arguments. Should we even include the variable level of arguments here?